wolfcrypt/src/port/autosar/test.c

```diff
@@ -0,0 +1,430 @@
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#ifdef HAVE_CONFIG_H
+    #include <config.h>
+#endif
+
+#include <wolfssl/wolfcrypt/settings.h>
+#include <wolfssl/wolfcrypt/logging.h>
+#include <wolfssl/wolfcrypt/port/autosar/Csm.h>
+#define BLOCK_SIZE 16
+
+static int singleshot_test(void)
+{
+    Std_ReturnType ret;
+
+    uint8 cipher[BLOCK_SIZE * 2];
+    uint8 plain[BLOCK_SIZE * 2];
+
+    uint32 cipherSz = 0;
+    uint32 plainSz  = 0;
+    const uint8 msg[] = { /* "Now is the time for all " w/o trailing 0 */
+        0x6e,0x6f,0x77,0x20,0x69,0x73,0x20,0x74,
+        0x68,0x65,0x20,0x74,0x69,0x6d,0x65,0x20
+    };
+    const uint8 verify[] =
+    {
+        0x95,0x94,0x92,0x57,0x5f,0x42,0x81,0x53,
+        0x2c,0xcc,0x9d,0x46,0x77,0xa2,0x33,0xcb
+    };
+    const uint8 key[] = "0123456789abcdef   ";
+    const uint8 iv[]  = "1234567890abcdef   ";
+
+    XMEMSET(cipher, 0, BLOCK_SIZE);
+    XMEMSET(plain, 0, BLOCK_SIZE);
+
+    /* set key that will be used for encryption */
+    ret = Csm_KeyElementSet(0U, CRYPTO_KE_CIPHER_KEY, key, BLOCK_SIZE);
+    if (ret != E_OK) {
+        printf("Issue with setting key");
+        return -1;
+    }
+
+    ret = Csm_KeyElementSet(1U, CRYPTO_KE_CIPHER_IV, iv, BLOCK_SIZE);
+    if (ret != E_OK) {
+        printf("Issue with setting key IV");
+        return -1;
+    }
+
+    /* encrypt data using AES CBC */
+    ret = Csm_Encrypt(1U, CRYPTO_OPERATIONMODE_SINGLECALL, msg, BLOCK_SIZE,
+        cipher, &cipherSz);
+    if (ret != E_OK) {
+        printf("Issue with encrypting msg");
+        return -1;
+    }
+
+    if (XMEMCMP(cipher, verify, BLOCK_SIZE) != 0) {
+        printf("Error with cipher data\n");
+        return -1;
+    }
+
+    /* set key that will be used for decryption */
+    ret = Csm_KeyElementSet(0U, CRYPTO_KE_CIPHER_KEY, key, BLOCK_SIZE);
+    if (ret != E_OK) {
+        printf("Issue with setting key");
+        return -1;
+    }
+
+    ret = Csm_KeyElementSet(1U, CRYPTO_KE_CIPHER_IV, iv, BLOCK_SIZE);
+    if (ret != E_OK) {
+        printf("Issue with setting key IV");
+        return -1;
+    }
+
+    /* decrypt data using AES CBC */
+    ret = Csm_Decrypt(2U, CRYPTO_OPERATIONMODE_SINGLECALL, cipher, BLOCK_SIZE,
+        plain, &plainSz);
+    if (ret != E_OK) {
+        printf("Issue with decrypting msg");
+        return -1;
+    }
+
+    if (XMEMCMP(msg, plain, BLOCK_SIZE) != 0) {
+        printf("Error with plain data\n");
+        return -1;
+    }
+
+    return 0;
+}
+
+
+static int update_test(void)
+{
+    Std_ReturnType ret;
+
+    uint8 cipher[BLOCK_SIZE * 3];
+    uint8 plain[BLOCK_SIZE * 3];
+
+    uint32 cipherSz = 0;
+    uint32 plainSz  = 0;
+    const uint8 msg[] = { /* "Now is the time for all " w/o trailing 0 */
+        0x6e,0x6f,0x77,0x20,0x69,0x73,0x20,0x74,
+        0x68,0x65,0x20,0x74,0x69,0x6d,0x65,0x20
+    };
+    const uint8 key[] = "0123456789abcdef   ";  /* align */
```

Comment:
What's the comment mean?

Reply:
left over from copy paste of keys that were used in wolfcrypt/test/test.c